server/query.go

```diff
@@ -38,74 +38,106 @@ func onlyOneSynonym(ops map[string]string, op1 string, op2 string) (string, erro
 	return ops[op2], nil
 }
 
-func ParseQuery(query string) (pb.Query, error) {
+func ParseQuery(query string, globalRegex bool) (pb.Query, error) {
+	var out pb.Query
+
 	ops := make(map[string]string)
 	key := ""
+	term := ""
 	q := strings.TrimSpace(query)
+	inRegex := globalRegex
+	justGotSpace := true
 
 	for {
 		m := pieceRE.FindStringSubmatchIndex(q)
 		if m == nil {
-			ops[key] += q
+			term += q
+			if _, alreadySet := ops[key]; alreadySet {
+				return out, fmt.Errorf("(a) got term twice: %s", key)
```

Comment:
Are these `(a)` and `(b)` for your debugging? Should we clean up these error message?